src/GNDStk/gnds-tree.hpp

```diff
@@ -0,0 +1,644 @@
+
+// -----------------------------------------------------------------------------
+// Tree (templated)
+// tree = Tree<>
+// -----------------------------------------------------------------------------
+
+template<
+   template<class...> class MCON, // metadata container
+   template<class...> class CCON  // children container
+>
+class Tree {
+   static std::string static_str1;
+   static std::string static_str2;
+
+public:
+
+   // ------------------------
+   // Data
+   // ------------------------
+
+   // root
+   std::shared_ptr<Node<MCON,CCON>> root;
```

Comment:
I can't think of any reasonable and non-convoluted way to implement a tree structure without using a pointer in some form. Technically it can be done, but it would be ugly.